src/core/server/metrics/metrics_service.ts

```diff
@@ -83,5 +92,31 @@ export class MetricsService
       clearInterval(this.collectInterval);
     }
     this.metrics$.complete();
+    this.subscription.unsubscribe();
+  }
+
+  private convertToMetricEvent(metrics: OpsMetrics): OpsMetricsEvent {
+    const { processes, process, ...rest } = metrics;
+    const formattedMetrics =
+      metrics.processes && metrics.processes?.length > 0
+        ? { ...rest, single_process: metrics.processes[0] }
+        : { ...rest, single_process: metrics.process };
+    // single node process
+    return { ...formattedMetrics };
+  }
+
+  private setupOpsMetricsEventType(
+    analytics: AnalyticsServiceSetup,
+    metricsObservable: Observable<OpsMetrics>
+  ) {
+    analytics.registerEventType<OpsMetricsEvent>({
+      eventType: 'core-ops_metrics',
+      schema: opsMetricsEventSchema,
+    });
+    this.subscription.add(
+      metricsObservable.subscribe((metrics: OpsMetrics) => {
+        analytics.reportEvent('core-ops_metrics', { ...this.convertToMetricEvent(metrics) });
```

Comment:
nit: the method already spreads twice. For performance reasons, we may want to not spread here as well?

Reply:
we have to spread here to overcome:
```
Argument of type 'OpsMetricsEvent' is not assignable to parameter of type 'Record<string, unknown>'.
  Index signature for type 'string' is missing in type 'OpsMetricsEvent'.
```